### PR TITLE
Assorted fixes for tasks and task API

### PR DESCRIPTION
**Pull request by jamalex**
## Summary

Content import/export should now work seamlessly via the API, and progress should be displayed correctly.

## Reviewer guidance

Even though the PR isn't huge overall, it may be productive to review commit-by-commit, as they're chunked into meaningful units.

## Issues addressed

- https://github.com/learningequality/kolibri/issues/489 (https://trello.com/c/QvLkPIzg/560-import-export-tasks-api-reports-random-completion-percentage)
- https://trello.com/c/N3sptoAQ/537-import-export-startremoteimport-api-call-should-synchronously-return-an-error-if-channel-doesn-t-exist
- Queued transfers were slow due to small block size (-> frequent ormq database saves)
- Task status constants on client didn't match those on server, so status wasn't displayed correctly.
- Exporting to external drive via API was broken

### CODE REVIEW

File: kolibri/tasks/management/commands/base.py [hunk 1]
```diff
@@ -1,45 +1,58 @@
...
-            message=message,
-            extra_data=extra_data,
-        )
+        self.message = message
+
+        self.extra_data = extra_data
+
+        if callable(self.update_callback):
+            p = self.get_progress()
+            self.update_callback(p.progress_fraction, p)
```
Comment (benjaoming) [inline kolibri/tasks/management/commands/base.py:46]: Why does this receive both the full `Progress` instance and a property of the same object instance? :)

File: kolibri/content/utils/transfer.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ class TransferNotYetClosed(Exception):
 
 class Transfer(object):
 
-    def __init__(self, source, dest, block_size=2048, remove_existing_temp_file=True):
+    def __init__(self, source, dest, block_size=2097152, remove_existing_temp_file=True):
```
Comment (aronasorman) [inline kolibri/content/utils/transfer.py:28]: 2 MB? Can you add a comment why this is increased?

File: kolibri/tasks/management/commands/base.py (hunk 1 above)
Comment (aronasorman) [reply kolibri/tasks/management/commands/base.py:46]: The intent, was that the 2nd argument is supposed to be any arbitrary object that we want to pass to the caller - the 1st argument is the progress in general.

File: kolibri/tasks/management/commands/base.py [hunk 3]
```diff
@@ -65,15 +78,20 @@ class AsyncCommand(BaseCommand):
 
     """
 
+    def __init__(self, *args, **kwargs):
+        self.progresstrackers = []
+
+    def _update_all_progress(self, progress_fraction, progress):
```
Comment (aronasorman) [inline kolibri/tasks/management/commands/base.py:84]: Can you add a comment explaining the algorithm + rationale?

File: kolibri/tasks/api.py [hunk 4]
```diff
@@ -44,7 +48,14 @@ def startremoteimport(self, request):
         if "channel_id" not in request.data:
             raise serializers.ValidationError("The 'channel_id' field is required.")
 
-        task_id = async(_networkimport, request.data['channel_id'], group=TASKTYPE, progress_updates=True)
+        channel_id = request.data['channel_id']
+
+        # ensure the requested channel_id can be found on the central server, otherwise error
+        status = requests.head(get_content_database_file_url(channel_id)).status_code
+        if status == 404:
+            raise Http404(_("The requested channel does not exist on the content server."))
```
Comment (indirectlylit) [inline kolibri/tasks/api.py:56]: nice

File: kolibri/tasks/api.py [hunk 5]
```diff
@@ -113,67 +124,41 @@ def localdrive(self, request):
...
     return {
-        "type": tasktype,
-        "status": status,
-        "percentage": percentage,
-        "metadata": task_type_specific_metadata,
-        "id": id,
+        "type": task_instance.group,
+        "status": task_instance.task_status,
+        "percentage": task_instance.progress_fraction,
+        "progress": [dict(p.__dict__) for p in task_instance.progress_data],
```
Comment (indirectlylit) [inline kolibri/tasks/api.py:162]: What is the form of the data in here?

Comment (indirectlylit) [reply kolibri/tasks/api.py:162]: I'm getting this error here:

```
'unicode' object has no attribute '__dict__'
```

Comment (aronasorman) [reply kolibri/tasks/api.py:162]: @indirectlylit I'll walk over there to help you debug it

Comment (aronasorman) [reply kolibri/tasks/api.py:162]: Prognosis: `progress_data` seems like it's getting pickled into one format, then base64'ed. Strange.

Comment (indirectlylit) [reply kolibri/tasks/api.py:162]: Aron committed a guard here: https://github.com/learningequality/kolibri/pull/530/commits/1cd0686aa623507a59e8dae669689331dec0ba40

Comment (indirectlylit) [reply kolibri/tasks/api.py:162]: (unfortunately, still get the same error)

**Pull request merged**